boring/src/ssl/mod.rs

```diff
@@ -703,31 +696,41 @@ impl SslContextBuilder {
         unsafe {
             init();
             let ctx = cvt_p(ffi::SSL_CTX_new(SslMethod::tls_with_buffer().as_ptr()))?;
+            let mut ctx = SslContextBuilder::from_ptr(ctx, true);
+
+            ctx.set_raw_public_key_mode();
 
-            Ok(SslContextBuilder::from_ptr(ctx, true))
+            Ok(ctx)
         }
     }
 
-    /// Sets raw public key certificate in DER format.
-    pub fn set_rpk_certificate(&mut self, cert: &[u8]) -> Result<(), ErrorStack> {
+    /// SSL_CTX_set_raw_public_key_mode indicates that |ctx| wishes to receive a raw
+    /// public key from a server rather than an X.509 certificate chain. Clients
+    /// should install a custom verifier with |SSL_CTX_set_custom_verify| or
+    /// |SSL_set_custom_verify| to verify the server's public key, which can be
+    /// obtained via |SSL_get0_peer_pubkey| or |SSL_get0_unparsed_peer_pubkey|. This
+    /// is only supported for client connections currently and only with TLSv1.3.
+    pub fn set_raw_public_key_mode(&mut self) {
         unsafe {
-            cvt(ffi::SSL_CTX_set_server_raw_public_key_certificate(
-                self.as_ptr(),
-                cert.as_ptr(),
-                cert.len() as u32,
-            ))
-            .map(|_| ())
+            ffi::SSL_CTX_set_raw_public_key_mode(self.as_ptr());
         }
     }
 
-    /// Sets RPK null chain private key.
-    pub fn set_null_chain_private_key<T>(&mut self, key: &PKeyRef<T>) -> Result<(), ErrorStack>
+    /// SSL_CTX_set_raw_public_key_and_key sets a raw public key (in
+    /// SubjectPublicKeyInfo format) and the corresponding private key for a TLS
+    /// server. References to the given |CRYPTO_BUFFER| and |EVP_PKEY| objects are
+    /// added as needed. Exactly one of |privkey| or |privkey_method| may be
+    /// non-NULL. |spki| must be non-NULL iff |privkey_method| is, otherwise the
+    /// public key will be generated from |privkey|. It returns one on success and
+    /// zero on error.
+    pub fn set_raw_public_key_and_key<T>(&mut self, key: &PKeyRef<T>) -> Result<(), ErrorStack>
```

Comment:
We need to mention here that this need to be used along with `set_custom_verify`. In addition we should introduce a convenience method that automatically creates a pub key from the provided private key and sets up a callback like you do in tests.